lib/spack/llnl/util/symlink.py

```diff
@@ -66,27 +68,27 @@ def _win32_junction(path, link):
 
 @lang.memoized
 def _win32_can_symlink():
-    tempdir = tempfile.mkdtemp()
+    tempdir = Path(tempfile.mkdtemp())
 
-    dpath = join(tempdir, "dpath")
-    fpath = join(tempdir, "fpath.txt")
+    dpath = tempdir / "dpath"
+    fpath = tempdir / "fpath.txt"
 
-    dlink = join(tempdir, "dlink")
-    flink = join(tempdir, "flink.txt")
+    dlink = tempdir / "dlink"
+    flink = tempdir / "flink.txt"
 
     import llnl.util.filesystem as fs
 
     fs.touchp(fpath)
 
     try:
         os.symlink(dpath, dlink)
```

Comment:
We can use [`Path.symlink_to()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.symlink_to) here:
```suggestion
        dlink.symlink_to(dpath, is_directory=True)
```

Reply:
Same comment as above. I should have avoided touching this file (outside of the initial symlink entrypoint) as all of my changes will likely be moot pretty shortly (or whenever #34701 lands)

Reply:
To be clear, you're saying that if you avoid touching `symlink.py`, you don't think #34701 is blocking this?

Reply:
Right. That being said, I'm not sure what the ETA is on that PR, so maybe we want to make these changes and have them in develop, and then handle the conflicts that causes later. I'll reach out to the PR authors and see where they're at.